core/instance_queries.go

```diff
@@ -120,7 +120,7 @@ func (i *instance) belongsToEnabledASG() bool {
 			asg.loadLaunchConfiguration()
 			asg.loadLaunchTemplate()
 			i.asg = &asg
-			i.price = i.typeInfo.pricing.onDemand
+			i.price = i.typeInfo.pricing.onDemand / i.region.conf.OnDemandPriceMultiplier * i.asg.config.OnDemandPriceMultiplier
```

Comment:
Did you test this?

We used to have an issue with this logic(it used to consider the square of the multiplier value instead of the actual value) which I tried to fix a few months ago, but the current fix only considers the parameter from the Lambda environment variables, not from the tag-based ASG overrides.

Reply:
right, but what if you also set the value as env var? I suspect it will be squared, because the value comes from both tag and env var and gets multiplied twice.

Reply:
yes, i suspect that too.
So we should add a condition to ignore `i.region.conf.OnDemandPriceMultiplier` if `i.asg.config.OnDemandPriceMultiplier` is != `1`.
Only one multiplier should be taken in account.
What do you think ?

Reply:
We calculate region instance ondemand price in method `determineInstanceTypeInformation`.
And we multiply it by the global `OnDemandPriceMultiplier`:
`price.onDemand = it.Pricing[r.name].Linux.OnDemand * cfg.OnDemandPriceMultiplier`

We call `determineInstanceTypeInformation` for both instance launch event and cron event.
So we need to divide by that global value before multiplying by the `OnDemandPriceMultiplier` ASG specific.

Reply:
thanks, I'll look into this over the next few days, I want to address this issue somehow.